### ISSUE TITLE
[Docs]: Resource: **aws_customer_gateway** _ip_address_ argument is marked as optional

### ISSUE THREAD

**Issue by peterbratu** [hashicorp/terraform-provider-aws#37553]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/customer_gateway#ip_address

### Description

Not specifying _ip_address_ parameter in the _aws_customer_gateway_ resource results in this error:
`Error: creating EC2 Customer Gateway: MissingParameter: The request must contain the parameter ipAddress`

### Would you like to implement a fix?

None

**Comment by CodeSinghh** [comment]
I have read this issue thoroughly and I want to solve this ✍️

**Comment by dthvt** [comment]
ip_address is optional. Customer Gateways using public key authentication instead of pre-shared key do not require static IP address.

See https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateCustomerGateway.html

**Comment by Bexanderthebex** [comment]
@dthvt that is true, so instead of this being an issue of documentation, this should probably be a bug request.
